### PR TITLE
[BE]: Enable ruff PIE794 and fix bugs it found in test suite

### PR DESCRIPTION
**Pull request by Skylion007**
Enables some tests that were incorrectly not being run and enables PIE794 globally. This rule checks if a classvar is defined twice as flags it as it is likely a bug. In fact, we found several cases where it was a bug. It does have a couple of false positives which I flagged upstream and replaced with noqas: https://github.com/astral-sh/ruff/issues/8497

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/torch_np/numpy_tests/core/test_multiarray.py [hunk 1]
```diff
@@ -3257,15 +3257,15 @@ class TestArgmax(TestCase):
         (np.array(d[0], dtype=t), d[1])
         for d, t in (itertools.product(usg_data, (np.uint8,)))
     ]
-    darr = darr + [
+    darr = darr + [  # noqa: PIE794
```
Comment (malfet) [inline test/torch_np/numpy_tests/core/test_multiarray.py:3260]: Lists are mutable in python ,aren't they? In that case, wouldn't it be better than adding `noqa:` suppression?
```suggestion
    darr += [
```

File: test/torch_np/numpy_tests/core/test_multiarray.py [hunk 2]
```diff
@@ -3257,15 +3257,15 @@ class TestArgmax(TestCase):
...
         (np.array(d[0], dtype=t), d[1])
         for d, t in (
             itertools.product(
                 sg_data, (np.int8, np.int16, np.int32, np.int64, np.float32, np.float64)
             )
         )
     ]
-    darr = darr + [
+    darr = darr + [  # noqa: PIE794
```
Comment (malfet) [inline test/torch_np/numpy_tests/core/test_multiarray.py:3268]: Same as above
```suggestion
    darr += [
```

File: test/torch_np/numpy_tests/core/test_multiarray.py [hunk 3]
```diff
@@ -3366,15 +3366,15 @@ class TestArgmin(TestCase):
         (np.array(d[0], dtype=t), d[1])
         for d, t in (itertools.product(usg_data, (np.uint8,)))
     ]
-    darr = darr + [
+    darr = darr + [  # noqa: PIE794
```
Comment (malfet) [inline test/torch_np/numpy_tests/core/test_multiarray.py:3369]:
```suggestion
    darr += darr + [
```

**Review by malfet — APPROVED** [review]
LGTM, but please consider replacing `darr = darr + [  # noqa` with `darr += [`

File: test/torch_np/numpy_tests/core/test_multiarray.py (hunk 1 above)
Comment (Skylion007) [reply test/torch_np/numpy_tests/core/test_multiarray.py]: Huh, that actually avoids the issue.

**Pull request closed without merging**